### ISSUE TITLE
[BUG][0.3.0][SAFARI] Blocked script execution in 'https://foo.bar' because the document's frame is sandboxed and the 'allow-scripts' permission is not set.

### ISSUE THREAD

**Issue by 88Maxwell** [Iterable/iterable-web-sdk#141]
# Problem 

In-app Iframe opened in Chrome and all links works correctly but in Safari I receive this error
```
Blocked script execution in 'https://foo.bar' because the document's frame is sandboxed and the 'allow-scripts' permission is not set.
```

<img width="467" alt="Screen Shot 2022-08-04 at 21 59 48" src="https://user-images.githubusercontent.com/24461597/182932343-fee58411-8106-4480-856e-bb69fe184759.png">

Code that I used to receive in-apps:
```js
export const requestInAppMessage = async () => {
  const { request } = getInAppMessages(
    {
      count           : 20,
      displayInterval : 1000,
      packageName     : ITERABLE_PACKAGE_NAME,
      rightOffset     : "120px",
      topOffset       : "120px",
      bottomOffset    : "120px",
      handleLinks     : "external-new-tab", // Do not help in this case.
    },
    { display: "immediate" },
  );
  const { data } = await request();
  return data;
};

```

### Versions 
- Safari -  Version 15.6 (17613.3.9.1.5)
- Google Chrome -  Version 103.0.5060.134 (Official Build) (x86_64)
- SDK  - version 0.3.0

### Current behaviour
In-app Iframe opened in Chrome and all links works correctly but in Safari I receive this error
```
Blocked script execution in 'https://foo.bar' because the document's frame is sandboxed and the 'allow-scripts' permission is not set.
```

### Expected behaviour
Links work correctly as like in Chrome./

### Steps to reproduce 

Submit any in-app with links to users using Safari.

### Potential solves

https://stackoverflow.com/a/30785417

**Comment by pauljung14** [comment]
Hi @88Maxwell, thank you for your feedback on the web-sdk. We've implemented a workaround in v0.4.0. As long as your links set the `target` value to `_blank` / `_top` / `_parent`, Safari will respect that decision and open the link in a new tab / same tab. The workaround takes this approach because of restrictions in Safari that prevent bound event handlers from running in a sandboxed iframe.

Please note that if you set the `handleLinks` parameter on the getMessages call, it will not override the `target` value set for the links in Safari (though it will override for other browsers). We are planning to develop a long-term solution for the Safari issue in which `handleLinks` parameter will override the link target settings the way it does for all other browsers, but we do not yet have a timeline for that. In the meantime, please feel free to employ the workaround.

**Comment by rasavage** [comment]
This is still an issue in v1.0.8.

> As long as your links set the target value to _blank / _top / _parent, Safari will respect that decision and open the link in a new tab / same tab.

From our tests in Safari, `_blank` doesn't work and is the uncustomisable `target` given to all button links by the template builder.

Iterable support say they have security concerns with `allow-scripts` on the sandboxed iframe so we have suggested using the `allow-popups-to-escape-sandbox` flag (as recommended by Google - [Allow popups to escape sandboxed iframe Sample](https://googlechrome.github.io/samples/allow-popups-to-escape-sandbox/)) that has 6+years of browser support.

> In order to mitigate the risks associated with ads when using sandboxed iframes, developers can now add "allow-popups-to-escape-sandbox" to the frame "attribute". This allows the sandboxed document to spawn new windows without forcing the sandboxing flags upon them, hence creating a clean browsing context. For instance, a third-party advertisement could be safely sandboxed without forcing the same restrictions upon a landing page.
> 
> Popup windows can be spawned with window.open() and target="_blank" links.

```diff
-    iframe.setAttribute('sandbox', 'allow-same-origin allow-popups allow-top-navigation');
+    iframe.setAttribute('sandbox', 'allow-same-origin allow-popups allow-top-navigation allow-popups-to-escape-sandbox');
```
We've had to apply the attached patch in the meantime:
[@iterable+web-sdk+1.0.8.patch](https://github.com/Iterable/iterable-web-sdk/files/14939286/%40iterable%2Bweb-sdk%2B1.0.8.patch)
